instantsearch/src/commonMain/kotlin/com/algolia/instantsearch/helper/searcher/internal/withUserAgent.kt

```diff
@@ -1,12 +1,17 @@
 package com.algolia.instantsearch.helper.searcher.internal
 
 import com.algolia.instantsearch.core.InstantSearch
+import com.algolia.instantsearch.helper.extension.telemetrySchema
 import com.algolia.search.dsl.requestOptions
 import com.algolia.search.transport.RequestOptions
 import io.ktor.http.HttpHeaders
 
-internal fun RequestOptions?.withUserAgent(): RequestOptions {
+internal fun RequestOptions?.withUserAgentTelemetry(): RequestOptions {
     return requestOptions(this) {
-        header(HttpHeaders.UserAgent, "$osVersion; ${InstantSearch.userAgent}")
+        header(HttpHeaders.UserAgent, userAgent(telemetrySchema()))
     }
 }
+
+private fun userAgent(telemetry: String? = null): String {
+    return if (telemetry == null) "$osVersion; ${InstantSearch.userAgent}" else "$osVersion; ${InstantSearch.userAgent}; $telemetry"
```

Comment:
A purely aesthetic remark: is it possible to rewrite it to something like
```kotlin
val baseUserAgent = "$osVersion; ${InstantSearch.userAgent}"
return telemetry?.let { baseUserAgent + "; $it" } ?: baseUserAgent
```

Reply:
Sure yes! :)